src/python/library/setup.py

```diff
@@ -69,15 +69,14 @@ def req_file(filename, folder="requirements"):
 extras_require = {
     'grpc': req_file("requirements_grpc.txt"),
     'http': req_file("requirements_http.txt"),
+    'cudashm': req_file("requirements_cudashm.txt"),
```

Comment:
I think since it is now pure Python, we can always include the CUDA shared memory util in the package, and convert the runtime check to inform "cuda-python" should be installed to use the package